src/backend/metal/src/device.rs

```diff
@@ -1250,11 +1281,8 @@ impl hal::Device<Backend> for Device {
         if self.private_caps.resource_heaps {
             // We don't know what memory type the user will try to allocate the buffer with, so we test them
             // all get the most stringent ones. Note we don't check Shared because heaps can't use it
-            for &options in [
-                MTLResourceOptions::StorageModeManaged,
-                MTLResourceOptions::StorageModeManaged | MTLResourceOptions::CPUCacheModeWriteCombined,
-                MTLResourceOptions::StorageModePrivate,
-            ].iter() {
+            for mt in &self.memory_types {
+                let options = map_memory_properties_to_options(mt.properties);
```

Comment:
What prevents `Shared` from being specified now? Or does above the comment no longer apply?

Reply:
eh, right, hmmm